odl/operator/oputils.py

```diff
@@ -152,33 +159,49 @@ def power_method_opnorm(op, niter, xstart=None):
                 ValueError('a starting element must be defined in case the '
                            'operator domain has no `one()`'), exc)
     else:
-        x = xstart.copy()
+        x = op.domain.element(xstart)
 
-    tmp = op.range.element()
+    x_norm = x.norm()
+    if x_norm == 0:
+        raise ValueError('input ``x=0``')
+    x /= x_norm
+
+    niter = int(niter)
+    if niter <= 0:
+        raise ValueError('niter not non-negative')
 
+    tmp = op.range.element()
     if use_normal:
         # Do the power iteration for A*A; the norm of A*A(x_N) is then
         # an estimate of the square of the operator norm
         # We do only half the number of iterations compared to the usual
         # case to have the same number of operator evaluations.
-        niter = ceil(niter / 2)
-        for _ in range(niter):
+        half_niter = niter // 2
+        if half_niter * 2 != niter:
+            raise ValueError('niter not an even number')
+
+        for _ in range(half_niter):
             op(x, out=tmp)
             op.adjoint(tmp, out=x)
-            x /= x.norm()
 
-        op(x, out=tmp)
-        op.adjoint(tmp, out=x)
-        return sqrt(x.norm())
+            x_norm = x.norm()
+            if x_norm == 0:
+                raise ValueError('reached ``x=0``')
```

Comment:
There is an test for exactly this further down. Basically when you have an operator like

```
[[0, 1],
 [0, 0]]
```

and if you start with `[1, 1]`, you then get `[1, 0]`, and then `[0, 0]`.